packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -15,6 +15,7 @@ import 'build.dart';
 class BuildIOSCommand extends BuildSubCommand {
   BuildIOSCommand() {
     usesTargetOption();
+    usesFlavorOption();
     usesPubOption();
     argParser.addFlag('debug',
       negatable: false,
```

Comment:
seems weird that this doesn't use addBuildModeFlags but that's not part of your patch